File: angry_professor/angry_professor.js
function main() {
  var t = parseInt(readLine());
  for (var a0 = 0; a0 < t; a0++) {
    var n_temp = readLine().split(' ');
    var n = parseInt(n_temp[0]);
    var k = parseInt(n_temp[1]);
    a = readLine().split(' ');
    a = a.map(Number);
    var present = 0;
    a.forEach(function(arrival) {
      if (arrival <= 0) {
        present += 1;
      }
    })
    if (present >= k) {
      console.log('NO');
    } else if (present < k) {
      console.log('YES');
    }
  }

}

File: angry_professor/angry_professor.py
#!/bin/python3

import sys


t = int(input().strip())
for a0 in range(t):
    n, k = input().strip().split(' ')
    n, k = [int(n), int(k)]
    a = [int(a_temp) for a_temp in input().strip().split(' ')]
    present = 0
    for x in a:
        if x <= 0:
            present += 1
    if present >= k:
        print('NO')
    elif present < k:
        print('YES')
